### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by piesieomane**
In this PULL REQUEST the following list below was implemented;

- Install `RSpec`.
- Write `test` before creating the class `solver`.
- Created the factorial method
- Created the reverse method
- Created the fizzbuzz method

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+require_relative '../solver'
+
+describe Solver do
+  solver = Solver.new
+  describe '#factorial' do
```
Comment (danmainah) [inline spec/solver_spec.rb:5]: - [ ] Kindly add a test for negative values where the factorial method will raise an exception as an expectation. This is a project requirement.
``hint``  `expect { resolver.factorial(-5) }.to raise_exception`

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @piesieomane and @Lizdev-05 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @piesieomane, and @Lizdev-05👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver_spec.rb (hunk 1 above)
Comment (Lizdev-05) [reply spec/solver_spec.rb:5]: Thank you for the review.

**Comment by Lizdev-05** [comment]
Thank you for reviewing and approving our code
